presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -87,75 +63,146 @@ public CassandraClusteringPredicatesExtractor(List<CassandraColumnHandle> cluste
         return TupleDomain.withColumnDomains(notPushedDown);
     }
 
-    private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColumnHandle> clusteringColumns, TupleDomain<ColumnHandle> predicates)
+    private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColumnHandle> clusteringColumns, TupleDomain<ColumnHandle> predicates, String cassandraVersion)
     {
         ImmutableMap.Builder<ColumnHandle, Domain> domainsBuilder = ImmutableMap.builder();
-        ImmutableList.Builder<Set<Object>> clusteringColumnValues = ImmutableList.builder();
+        ImmutableList.Builder<String> clusteringColumnSql = ImmutableList.builder();
         for (CassandraColumnHandle columnHandle : clusteringColumns) {
             Domain domain = predicates.getDomains().get().get(columnHandle);
-
             if (domain == null) {
                 break;
             }
-
             if (domain.isNullAllowed()) {
-                return new ClusteringPushDownResult(domainsBuilder.build(), ImmutableSet.of());
+                break;
             }
-
-            Set<Object> values = domain.getValues().getValuesProcessor().transform(
-                    ranges -> {
-                        ImmutableSet.Builder<Object> columnValues = ImmutableSet.builder();
-                        for (Range range : ranges.getOrderedRanges()) {
-                            if (!range.isSingleValue()) {
-                                return ImmutableSet.of();
+            String predicateString = null;
+            predicateString = domain.getValues().getValuesProcessor().transform(ranges -> {
+                List<Object> singleValues = new ArrayList<>();
+                List<String> rangeConjuncts = new ArrayList<>();
+                String predicate = null;
+
+                for (Range range : ranges.getOrderedRanges()) {
+                    if (range.isAll()) {
+                        return null;
```

Comment:
Yes, in the case of the column has `range.all` then we should restrict the predicate till the previous column, I have added the condition to break the loop there.